streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroStreamsSnapshotDeserializer.java

```diff
@@ -65,7 +65,10 @@ private Object convertValue(Object deserializedObj) {
             List<Schema.Field> fields = indexedRecord.getSchema().getFields();
             ImmutableMap.Builder<String, Object> keyValues = ImmutableMap.builder();
             for (Schema.Field field : fields) {
-                keyValues.put(field.name(), convertValue(indexedRecord.get(field.pos())));
+                Object currentValue = convertValue(indexedRecord.get(field.pos()));
+                if (currentValue != null) {
```

Comment:
@priyank5485 if I understand correctly what @satishd is saying is we might have a schema where the filed might look like this
`driverId: int, null`
With the above field we can have a event with driverId = 101 and another event where driverId = null.
It looks like you are removing the driverId from streamlineEvent all together if its null. So in this case if we have a rules processor where user added a rule that says "driverId > 100". This could cause an issue as the user assumes there will be a filed called driverId in the event where as with this fix we are removing when its null. So a Map lookup for the driverId might not be successful when its null. 
I think it makes sense to keep the driverId even if its null.  Basically we are not being true to the schema thats expressed and can have issues since we are not meeting the expectation that the StreamlineEvent will contain all the keys expressed in a schema.
This also means users need to worry about null values coming in. If they wrote a generic custom processor they have to make sure the field that they are looking is not null before doing any operations on it.